obspy/io/segy/tests/test_core.py

```diff
@@ -350,7 +351,8 @@ def test_writing_using_core(self):
                 with NamedTemporaryFile() as tf2:
                     out_file2 = tf2.name
                     # Write twice and catch header warnings
-                    with warnings.catch_warnings(record=True):
+                    with warnings.catch_warnings():
```

Comment:
Can this be replaced with `pytest.warns` or `obspy.core.utils.testing.WarningCapture`?

Reply:
This is an interesting thing. `pytest.warns()` is advertised in pytest documentation, but raises an error for my local pytest version (6.3) (missing expected warning category). `pytest.warns(None)` is working for pytest 6.3, but is deprecated since pytest 7.0. `pytest.warns(UserWarning)` and `pytest.warns(SpecialSEGYWarning)` (dont remeber the correct warning name) are not catching the warning, although SpecialSEGYWarning is a child of UserWarning. I think its a bug in pytest. Same below. Let's just leave it as it is now.